src/vault/authenticate.clj

```diff
@@ -132,3 +132,35 @@
            :content-type :json
            :accept :json
            :as :json})))))
+
+
+(defmethod authenticate* :aws
```

Comment:
Read up on the docs a bit, and it looks like this is supporting the IAM authentication method, which goes through the AWS backend. I wonder if it's worth calling this `:iam` to be more specific, or if we should keep these 1:1 with backends. 🤔 No strong feeling one way or the other here.

Reply:
Perhaps `:aws-iam` since its specific?